cdc/owner.go

```diff
@@ -428,15 +429,31 @@ func (o *Owner) newChangeFeed(
 	}
 	errCh := make(chan error, 1)
 
-	primarySink, err := sink.NewSink(ctx, id, info.SinkURI, filter, info.Config, info.Opts, errCh)
+	// primarySink, err := sink.NewSink(ctx, id, info.SinkURI, filter, info.Config, info.Opts, errCh)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	// new asyncSink
+	cdcCtx := cdcContext.NewContext(ctx, &cdcContext.GlobalVars{})
+	cdcCtx = cdcContext.WithChangefeedVars(cdcCtx, &cdcContext.ChangefeedVars{
+		ID:   id,
+		Info: info,
+	})
+	primaryAsyncSink, err := newAsyncSink(cdcCtx)
```

Comment:
What about using cdc/owner/async_sink directly?

Reply:
I will try this.

Reply:
I made some changes on async_sink, so I think it’s better to use it separately from `cdc/owner/async_sink`.